lib/src/command/unpack.dart

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:path/path.dart' as p;
+import 'package:pub_semver/pub_semver.dart';
+import 'package:yaml/yaml.dart';
+
+import '../command.dart';
+import '../command_runner.dart';
+import '../entrypoint.dart';
+import '../io.dart';
+import '../log.dart' as log;
+import '../package_name.dart';
+import '../pubspec.dart';
+import '../sdk.dart';
+import '../solver/type.dart';
+import '../source/hosted.dart';
+import '../utils.dart';
+
+/// Handles the `deps` pub command.
+class UnpackCommand extends PubCommand {
+  @override
+  String get name => 'unpack';
+
+  @override
+  String get description => '''
+Downloads a package and unpacks it in place.
+
+For example:
+
+  $topLevelProgram pub unpack foo
+
+Downloads and extracts the lastest stable package:foo from pub.dev in a
+directory `foo-<version>`.
+
+  $topLevelProgram pub unpack foo:1.2.3-pre
+
+Downloads and extracts package:foo version 1.2.3-pre in a directory
+`foo-1.2.3-pre`.
+
+  $topLevelProgram pub unpack foo --output=archives
+
+Downloads and extracts latest stable version of package:foo in a directory
+`archives/foo-<version>`.
+
+  $topLevelProgram pub unpack 'foo:{hosted:"https://my_repo.org"}'
+
+Downloads and extracts latest stable version of package:foo from my_repo.org
+in a directory `foo-<version>`.
+
+
+Will resolve dependencies in the folder unless `--no-resolve` is passed.
+''';
+
```

Comment:
Do you want to include something in the example about a `resolve` flag?

Reply:
Done